### ISSUE TITLE
Odd behaviour for operator presedence in recent typescript (2.4.0)

### ISSUE THREAD

**Issue by janerivi** [Microsoft/TypeScript#16764]
**TypeScript Version:**  2.4.0 

**Code**
```
var numbers:any[] = ["2","10","3"]
var decimal = numbers[0] * 1 + numbers[1] / 60 + (numbers[2] + "." + (numbers[3] || 0)) as any / 3600; 
```

// A *self-contained* demonstration of the problem follows...

**Expected behavior:**

I would have expected  thre result to be:

```
var numbers = ["2", "10", "3"];
var decimal = numbers[0] * 1 + numbers[1] / 60 + (numbers[2] + "." + (numbers[3] || 0)) / 3600;
```

**Actual behavior:**

this transpiles surpricingly to me to: 
```
var numbers = ["2", "10", "3"];
var decimal = (numbers[0] * 1 + numbers[1] / 60 + (numbers[2] + "." + (numbers[3] || 0))) / 3600;
```
an extra and error causing parentesis was wrapped from the start of the expression to the place where "as any" was used.

This broke the code for us as we upgraded from an earlier version of typescript to 2.4.0

if I remove the any type from the array and remove "as any"  from the original code it works out just fine

**Comment by janerivi** [comment]
If I go the extra length and wrap the the parenthesis around the part I wuld like it to apply to it works
var decimal = numbers[0] * 1 + numbers[1] / 60 + ((numbers[2] + "." + (numbers[3] || 0)) as any) / 3600; 

But should that really be required when the highest presedence operator is already used for the part that I wanted to type "as any"?

**Comment by andy-ms** [comment]
`as` has the same precedence as `in`. So everything to the left of it is grouped together (then divided by 3600).
I think you need `(numbers[2] + "." + (numbers[3] || 0) as any)` instead of `(numbers[2] + "." + (numbers[3] || 0)) as any`.

**Comment by DanielRosenwasser** [comment]
It appears 2.3 had the same emit. What version are you upgrading from?

**Comment by janerivi** [comment]
1.8 or 1.6 i guess... I upgraded the visual studio tools and updated the tsc systemwide via npm - it was not a local dependency inside the project. But still I am confused why it is sensible to break backward compatibility by not respecting parenthesis "( )" presedence (wich I thought was the highest presedence in javascript).

**Comment by andy-ms** [comment]
The `as` expression is not enclosed in parentheses in your example. If you do it will be fixed.

**Comment by janerivi** [comment]
andy-ms: I knew before I posted the issue how to do a workaround by for example wrapping parentheses. The reason I posted the issue was:
1) That the  transpiling in recent typescript versions broke our legacy typescript transpiling so that our app got a bug where we hadnt changed a single character in the sourcecode. It did cost me a bit of time identifying where the bug originated (in the transpilation step).
2) I find the current way that typescript transpiles the original code odd.  I don't see why it is reasonable that an "as" operator right next to an enclosed parenthesis should need to affect beyond that parenthesis. I would expect a lot of other developers to have been surpriced and have had to spend time trying to figure out that bugs in their code was caused by the transpiler.
